src/group/group_context.rs

```diff
@@ -7,26 +12,65 @@
         epoch: GroupEpoch,
         tree_hash: Vec<u8>,
         confirmed_transcript_hash: Vec<u8>,
-        extensions: &[Box<dyn Extension>],
+        extensions: Vec<Box<dyn Extension>>,
     ) -> Result<Self, CodecError> {
         let mut group_context = GroupContext {
             group_id,
             epoch,
             tree_hash,
             confirmed_transcript_hash,
-            extensions: extensions.to_vec(),
+            extensions,
             serialized: vec![],
         };
         let serialized = group_context.encode_detached()?;
         group_context.serialized = serialized.to_vec();
         Ok(group_context)
     }
+
+    /// Create the group context of a new epoch from the one of the previous
+    /// epoch.
+    pub(crate) fn from_previous_group_context(
```

Comment:
`from_group_context`?
Also it's not entirely clear how this is a `from`, but requiring additional arguments.